### ISSUE TITLE
[nodemon] Internal watch failed: watch EMFILE

### ISSUE THREAD

**Issue by rjcoupe** [remy/nodemon#731]
Not sure what's prompted this, since the same operation was working fine a few days ago. I've rebooted to no effect, so presumably it's not ye olde too-many-open-files problem.

```
$ nodemon --debug --dump server.js
[nodemon] 1.8.1
[nodemon] to restart at any time, enter `rs`
[nodemon] watching: *.*
--------------
node: v0.10.26
nodemon: 1.8.1
command: node /Users/rcoupe/.nvm/v0.10.26/bin/nodemon --debug --dump server.js
cwd: /Users/rcoupe/Code/overlay-management-system
OS: darwin x64
--------------
{ run: false,
  system: { cwd: '/Users/rcoupe/Code/overlay-management-system' },
  required: false,
  dirs: [ '/Users/rcoupe/Code/overlay-management-system' ],
  timeout: 1000,
  options:
   { dump: true,
     ignore:
      [ '.git',
        'node_modules',
        'bower_components',
        '.sass-cache',
        re: /\.git|node_modules|bower_components|\.sass\-cache/ ],
     watch: [ '*.*', re: /.*\..*/ ],
     ignoreRoot: [ '.git', 'node_modules', 'bower_components', '.sass-cache' ],
     restartable: 'rs',
     colours: true,
     execMap: { py: 'python', rb: 'ruby' },
     stdin: true,
     runOnChangeOnly: false,
     verbose: false,
     stdout: true,
     execOptions:
      { script: 'server.js',
        exec: 'node',
        args: [ '--debug' ],
        scriptPosition: 1,
        nodeArgs: undefined,
        ext: 'js,json',
        env: {},
        execArgs: [] },
     monitor:
      [ '*.*',
        '!/Users/rcoupe/Code/overlay-management-system/.git/**/*',
        '!/Users/rcoupe/Code/overlay-management-system/node_modules/**/*',
        '!bower_components',
        '!.sass-cache' ] },
  load: [Function],
  reset: [Function: reset],
  lastStarted: 0,
  loaded: [],
  watchInterval: null,
  command:
   { raw: { executable: 'node', args: [ '--debug', 'server.js' ] },
     string: 'node --debug server.js' } }
--------------
```

**Comment by pengqingbiao** [comment]
+1

**Comment by joshcorn** [comment]
+1

**Comment by nwkeeley** [comment]
I am having this problem seemingly out of no where..... I restarted my machine but still not working

**Comment by rjcoupe** [comment]
Updating NPM hasn't fixed it for me. Nodemon works fine on other projects ..

```
$ npm install -g npm
npm http GET https://registry.npmjs.org/npm
npm http 200 https://registry.npmjs.org/npm
npm http GET https://registry.npmjs.org/npm/-/npm-3.5.0.tgz
npm http 200 https://registry.npmjs.org/npm/-/npm-3.5.0.tgz
~/.nvm/v0.10.26/bin/npm -> ~/.nvm/v0.10.26/lib/node_modules/npm/bin/npm-cli.js
npm@3.5.0 ~/.nvm/v0.10.26/lib/node_modules/npm

$ nodemon server.js
[nodemon] 1.8.1
[nodemon] to restart at any time, enter `rs`
[nodemon] watching: *.*
[nodemon] starting `node server.js`
[nodemon] Internal watch failed: watch EMFILE

$ npm -v
3.5.0
```

**Comment by flyingace** [comment]
I was having this problem as well. No idea if this will help anyone else or not but I went into my /usr/local/bin directory and removed nodemon with `rm -rf`. I had tried `npm uninstall nodemon` but that was returning me a message saying `npm WARN uninstall not installed in...` which makes me wonder whether the first install went wrong somehow or whether I'd previously installed it incorrectly and there was some kind of conflict?

Long story short, I removed the old nodemon and ran `npm install -g nodemon` again and that seems to have cleared the whole matter up.

**Comment by skalfyfan** [comment]
I'm getting a similar issue. Using Node 5.1.0 & NPM 3.3.12.

    root@014b87ca92f1:/var/www/html# nodemon app.js
    [nodemon] 1.8.1
    [nodemon] to restart at any time, enter `rs`
    [nodemon] watching: *.*
    ...
    info: Server lifted in `/var/www/html`
    info: To see your app, visit http://localhost:1337
    info: To shut down Sails, press <CTRL> + C at any time.

    debug: --------------------------------------------------------
    debug: :: Mon Dec 07 2015 16:54:53 GMT+0000 (UTC)

    debug: Environment : development
    debug: Port        : 1337
    debug: --------------------------------------------------------
    [nodemon] Internal watch failed: watch /var/www/html/node_modules/sails/node_modules/skipper/node_modules/multiparty/.travis.yml ENOSPC

It's not always the same file. It's a random different file every time but after about 10 seconds I always get the *[nodedemon] Internal watch failed* message and nodemon crashes? 

I'm running this in a Docker container. Node, NPM, Nodemon are all installed globally.
I've tried upgrading NPM to 3.5.1 and that did not fix the issue. Still crashes after ~10 seconds?

**Comment by nikitalarionov** [comment]
Try to use --watch "./directory". The latest node versions has permissions to number of open files.

**Comment by skalfyfan** [comment]
My related issue above (re: ENOSPC) was resolved by following [this](https://github.com/remy/nodemon/issues/731#issuecomment-162919977). Something apparently inherent in unix file systems and the number of files allowed to be watched at a time?

**Comment by jlone** [comment]
I run nodemon through screen and when this happened to me I fixed it by killing the screen session and starting a new one.

**Comment by zithan** [comment]
just run this to changed max_user_watchs--------->echo fs.inotify.max_user_watches=582222 | sudo tee -a /etc/sysctl.conf && sudo sysctl -p

**Comment by djensen47** [comment]
I'm seeing this issue consistently on the new Windows Subsystem for Linux.

**Comment by JensVerneuer** [comment]
@djensen47 actually the error (at least for me) on Windows Ubuntu shell is EPERM and not EMFILE which is discussed here.

**Comment by Carpetfizz** [comment]
Did anyone find a fix for the EPERM issue that occurs in the Windows 10 Bash?

**Comment by Almenon** [comment]
@Carpetfizz, I just had the same issue w/ Windows 10 Bash.  nodemon 1.10.0, node 6.4.0.
The error happens even if I am just watching one file (ex: nodemon -w app.js)

**Comment by aalexgabi** [comment]
@Carpetfizz For me it was the current path was not including the script I was running:
```
$ cd /
$ nodemon /tmp/myproject/bin/myproject.js
```
FIxed it with a cd to the folder containing the main script

Note: I use Ubuntu.

**Comment by EmersonStark** [comment]
killall -9 node resolve this problem

**Comment by Carpetfizz** [comment]
Thanks for the replies! I now use macOS, but I'll check out your solutions if I ever have to use W10Bash again.

**Comment by phuongleduy** [comment]
you can use ps -ef | grep node to find the process id

and then sudo kill -9 <PID>

**Comment by meghjoshi** [comment]
Run this command

`echo fs.inotify.max_user_watches=582222 | sudo tee -a /etc/sysctl.conf && sudo sysctl -p`

**Comment by katemicse** [comment]
i got error 
```
sysctl: illegal option -- p
usage: sysctl [-bdehiNnoqx] name[=value] ...
       sysctl [-bdehNnoqx] -a

```

**Comment by malec** [comment]
I was running in docker, forgot to set workdir - this fixed it.

**Comment by ramytamer** [comment]
I solved it on my mac by deleting the `node_modules` and re-installing `nodemon` again through it _not use it globally_; worked normally after that.

**Comment by noddycha** [comment]
Even i had the same error:

> [nodemon] Internal watch failed: EMFILE: too many open files, watch

I fixed it by just uninstalling and reinstalling nodemon:

`npm uninstall -g nodemon
npm install -g nodemon`

Hope this works for you guys as well 👍

**Comment by c3lvice** [comment]
You can try `--ignore node_modules`

**Comment by rumon5h** [comment]
In my case, 

Error: [nodemon] Internal watch failed: EMFILE: too many open files, watch 

Solution:
1. I close the vs code, 
2. Reopen it, and solve the problem.

**Comment by henrique1996n1** [comment]
For me closing and reopening the VS code worked too.
